### PR TITLE
Sanitation setup

### PR DESCRIPTION
**Pull request by rakaar**
Pylama, isort and mypy are setup. The same tools are used in pre-commit hook which can be checked at `.pre-commit-config.yaml` file. These tools are set up with travis CI which can be checked  at `.travis.yml` file.

concerns #7

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+repos:
```
Comment (thealphadollar) [inline .pre-commit-config.yaml]: This is something new and looks dope 😉 I'll take some time to review this.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+language: python
+python: -"3.6"
```
Comment (thealphadollar) [inline .travis.yml]: Let's use 3.7

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+# install the dependencies
+install:
+  - pip install -r requirements.txt
+
+# run the tests
+script:
+  - pylama
+  - isort -rc --check-only
+  - mypy --check-untyped-defs
```
Comment (thealphadollar) [inline .travis.yml]: create a `mypy.ini` config file?

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (thealphadollar) [inline .vscode/settings.json]: Please remove all instances of `.vscode`

File: Pipfile [hunk 5]
```diff
@@ -21,6 +21,10 @@ pytz = "==2019.1"
 six = "==1.12.0"
 sqlalchemy = "==1.3.5"
 werkzeug = "==0.15.4"
+pylama = "*"
```
Comment (thealphadollar) [inline Pipfile]: Please create config files for the sanity checkers as well.

**Comment by thealphadollar** [comment]
Please setup logging as well :D

File: .travis.yml (hunk 2 above)
Comment (rakaar) [reply .travis.yml]: should there be any change is `Pipfile` , where  it is [requires]python_version="3.6"

**Comment by thealphadollar** [comment]
Yes :)

**Comment by rakaar** [comment]
1)Removed the .vscode folder
2)changed Python version to 3.7 in `Pipfile` and `.travis.yml` file
3)Conifguration files for sanity tests:
- pylama : `pylama.ini` (as in README.md of pylama project)
- isort: `.isort.cfg` (as in README.md of isort project)
- mypy: `mypy.ini` (as in Sample Platform of CC Extractor)

File: .travis.yml (hunk 2 above)
Comment (thealphadollar) [reply .travis.yml]: Thanks.

File: .vscode/settings.json (hunk 4 above)
Comment (thealphadollar) [reply .vscode/settings.json]: Thanks :D

File: .travis.yml [hunk 6]
```diff
@@ -1,5 +1,7 @@
+dist: xenial
 language: python
-python: -"3.6"
+python: -"3.7"
+  -"nightly"
```
Comment (thealphadollar) [inline .travis.yml]: We don't need to test on nightly.

File: .isort.cfg [hunk 7]
```diff
@@ -0,0 +1,14 @@
+[settings]
+line_length=120
+force_to_top=file1.py,file2.py
```
Comment (thealphadollar) [inline .isort.cfg]: We do not require anything below line 2 (don't remove line 2) at the present.

File: pylama.ini [hunk 8]
```diff
@@ -0,0 +1,5 @@
+[pylama]
+format = pylint
+skip = */.tox/*,*/.env/*
+linters = pylint,mccabe
+ignore = F0401,C0111,E731
```
Comment (thealphadollar) [inline pylama.ini]: Remove line number 5

File: Pipfile (hunk 5 above)
Comment (thealphadollar) [reply Pipfile]: Thanks.

**Comment by thealphadollar** [comment]
@rakaar Can you please add information about this in a new file CONTRIBUTING.md as well? 

For an example on how to write about them, see Etiquettes under Contributing [here in sample-platform README.md](https://github.com/CCExtractor/sample-platform/blob/master/README.md)

Also add the instructions that one should run `pre-commit install` as the first thing after installing dependencies.

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+# install the dependencies
+install:
+  - pip install -r requirements.txt
+
+# run the tests
+script:
+  - pylama
+  - isort -rc --check-only
+  - mypy --check-untyped-defs
```
Comment (thealphadollar) [inline .travis.yml]: Please set --check-untyped-defs in mypy config itself and do not put anything here.

**Comment by rakaar** [comment]
@thealphadollar , Can you tell more about what should be present in `Contributing.md` for this proj

File: .travis.yml (hunk 9 above)
Comment (thealphadollar) [reply .travis.yml]: @rakaar Implement this please.

File: .mypy_cache/3.7/_ast.meta.json [hunk 10]
```diff
@@ -0,0 +1 @@
...08c270f478877c1f0148", "mtime": 1562042558, "options": {"allow_redefinition": false, "allow_untyped_globals": false, "always_false": [], "always_true": [], "bazel": false, "check_untyped_defs": false, "disallow_any_decorated": false, "disallow_any_explicit": false, "disallow_any_expr": false, "disallow_any_generics": false, "disallow_any_unimported": false, "disallow_incomplete_defs": false, "disallow_subclassing_any": false, "disallow_untyped_calls": false, "disallow_untyped_decorators": false, "disallow_untyped_defs": false, "follow_imports": "normal", "follow_imports_for_stubs": false, "ignore_errors": false, "ignore_missing_imports": true, "implicit_reexport": true, "local_partial_types": false, "mypyc": false, "new_semantic_analyzer": false, "no_implicit_optional": false, "platform": "linux", "plugins": [], "show_none_errors": true, "strict_equality": true, "strict_optional": true, "strict_optional_whitelist": null, "warn_no_return": true, "warn_return_any": false, "warn_unused_ignores": false}, "path": "/home/rka/.cache/pre-commit/repogzd7uevu/py_env-python3.6m/lib/python3.6/site-packages/mypy/typeshed/stdlib/3/_ast.pyi", "size": 7915, "suppressed": [], "version_id": "0.711"}
```
Comment (thealphadollar) [inline .mypy_cache/3.7/_ast.meta.json]: Please add `.mypy_cache` to .gitignore

**Comment by thealphadollar** [comment]
@rakaar I've attached a template contributing.md you can use. Our main purpose should be to convey what all standard we follow in the code as well as the organisation. You should mention that our code passes through x, y and z checks (in bullet form as in template) and what they ensure. Also, we should mention how we ensure that these are enforced (using pre-commit hook and CI tests).

Use your own creativity and I'll improve it when you send in the PR :smile: 

https://gist.github.com/PurpleBooth/b24679402957c63ec426

**Comment by thealphadollar** [comment]
Thank you. Looks great to me :D

**Pull request merged**